runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/ExecutableStageDoFnOperator.java

```diff
@@ -0,0 +1,211 @@
+    SdkHarnessClient.BundleProcessor<InputT> processor = client.getProcessor(
+            processBundleDescriptor.getProcessBundleDescriptor(), destination);
+    processor.getRegistrationFuture().toCompletableFuture().get();
+    Map<BeamFnApi.Target, Coder<WindowedValue<?>>> outputCoders =
+            processBundleDescriptor.getOutputTargetCoders();
+    ImmutableMap.Builder<BeamFnApi.Target,
+            SdkHarnessClient.RemoteOutputReceiver<?>> receiverBuilder = ImmutableMap.builder();
+    final Object collectorLock = new Object();
+    for (Map.Entry<BeamFnApi.Target, Coder<WindowedValue<?>>> entry : outputCoders.entrySet()) {
+      BeamFnApi.Target target = entry.getKey();
+      Coder<WindowedValue<?>> coder = entry.getValue();
+      SdkHarnessClient.RemoteOutputReceiver<WindowedValue<?>> receiver =
+              new SdkHarnessClient.RemoteOutputReceiver<WindowedValue<?>>() {
+                @Override
+                public Coder<WindowedValue<?>> getCoder() {
+                  return coder;
+                }
+
+                @Override
+                public FnDataReceiver<WindowedValue<?>> getReceiver() {
+                  return new FnDataReceiver<WindowedValue<?>>() {
+                    @Override
+                    public void accept(WindowedValue<?> input) throws Exception {
+                      logger.finer(String.format("Receiving value: %s", input));
+                      // TODO: Can this be called by multiple threads? Are calls guaranteed to
+                      // at least
+                      // be serial? If not, these calls may need to be synchronized.
+                      // TODO: If this needs to be synchronized, consider requiring immutable maps.
+                      synchronized (collectorLock) {
+                        // TODO: Get the correct output union tag from the corresponding output tag.
+                        // Plumb through output tags from process bundle descriptor.
+                        // NOTE: The output map is guaranteed to be non-empty at this point,
+                        // so we can
+                        // always grab index 0.
+                        // TODO: Plumb through TupleTag <-> Target mappings to get correct union tag
+                        c.output(input);
+                      }
+                    }
+                  };
+                }
+              };
+      receiverBuilder.put(target, receiver);
+    }
+    Map<BeamFnApi.Target,
+            SdkHarnessClient.RemoteOutputReceiver<?>> receiverMap =
+            receiverBuilder.build();
+
+    SdkHarnessClient.ActiveBundle<InputT> bundle = processor.newBundle(receiverMap);
+    try (CloseableFnDataReceiver<WindowedValue<InputT>> inputReceiver = bundle.getInputReceiver()) {
+      logger.finer(String.format("Sending value: %s", c.element()));
+      // TODO: check that we are actually getting WindowedValue here
+      inputReceiver.accept((WindowedValue<InputT>) c.element());
```

Comment:
Instead of providing `@DoFn.ProcessElement`, which does not have access to `WindowedValue`, we need to override the DoFnRunner via `createWrappingDoFnRunner`